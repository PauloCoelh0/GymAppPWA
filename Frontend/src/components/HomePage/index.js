import React, { useState } from "react";
import "./styles.module.scss";
import { Row, Col, Container } from "reactstrap";
import Header from "../Header";

const HomePage = () => {
  return (
    <html>
      <head>
        <title>My Gym</title>
      </head>
      <body>
        <h1>Welcome to My Gym</h1>
        <p>
          Here at My Gym, we offer a variety of membership options to fit your
          fitness needs and budget.
        </p>

        <h2>Membership Options</h2>
        <table>
          <tr>
            <th>Membership Type</th>
            <th>Price per Month</th>
          </tr>
          <tr>
            <td>Basic</td>
            <td>$20</td>
          </tr>
          <tr>
            <td>Standard</td>
            <td>$50</td>
          </tr>
          <tr>
            <td>Premium</td>
            <td>$80</td>
          </tr>
        </table>

        <h3>Basic Membership</h3>
        <p>
          Our basic membership includes access to all of our cardio equipment
          and weight machines.
        </p>

        <h3>Standard Membership</h3>
        <p>
          In addition to basic membership benefits, our standard membership
          includes access to group fitness classes and a personal training
          session per month.
        </p>

        <h3>Premium Membership</h3>
        <p>
          Our premium membership includes all basic and standard membership
          benefits, as well as 24/7 access to the gym and access to our sauna
          and steam room.
        </p>

        <p>Visit us today and start reaching your fitness goals!</p>
      </body>
    </html>
  );
};

export default HomePage;
