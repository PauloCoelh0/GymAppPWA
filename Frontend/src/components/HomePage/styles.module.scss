body {
    font-family: Arial, sans-serif;
    margin: 0;
    padding: 0;
    background-color: #f2f2f2;
  }
  h1, h2, h3 {
    text-align: center;
  }
  /* Table styles */
  table {
    border-collapse: collapse;
    margin: 50px auto;
  }
  th, td {
    border: 1px solid #dddddd;
    padding: 8px;
    text-align: left;
  }
  th {
    background-color: #dddddd;
  }
  /* Highlight the first row of the table */
  tr:first-child th {
    background-color: #4CAF50;
    color: white;
  }
  /* Add some hover effect to the table rows */
  tr:hover {
    background-color: #f5f5f5;
  }
  /* Add some padding and margin to the h1, h2 and h3 tags */
  h1 {
    padding-top: 50px;
    margin-bottom: 30px;
    font-size: 3em;
    color: #4CAF50;
  }
  h2 {
    margin-bottom: 30px;
    font-size: 2em;
  }
  h3 {
    margin-bottom: 30px;
    font-size: 1.5em;
  }
  /* Add some margin and padding to the p tags */
  p {
    text-align: center;
    margin: 20px 0;
    padding: 0 50px;
    font-size: 1.2em;
  }