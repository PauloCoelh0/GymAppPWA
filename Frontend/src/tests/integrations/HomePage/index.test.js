import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import HomePage from "../../../components/HomePage";

describe("HomePage", () => {
  it("renders corretly the component", () => {
    const { container } = render(<HomePage />);

    expect(container).toMatchSnapshot();
  });

  it("renders the login form correctly", () => {
    render(<HomePage />);

    expect(screen.getByText("Admin")).toBeInTheDocument();
  });

  describe("when click the button Login with Qr Code", () => {
    it("renders the component QrRead", () => {
      render(<HomePage />);

      userEvent.click(screen.getByText("Login with Qr Code"));

      expect(screen.getByText("QrCode")).toBeInTheDocument();
    });
  });
});
