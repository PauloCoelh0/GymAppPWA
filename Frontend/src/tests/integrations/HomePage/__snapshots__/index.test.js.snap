// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders corretly the component 1`] = `
<div>
  <div>
    <main />
  </div>
</div>
`;
